### PR TITLE
bpo-36686: Improve the documentation of the std* params in loop.subprocess_exec

### PR DESCRIPTION
**Pull request by sbstp**
https://bugs.python.org/issue36686

### CODE REVIEW

File: Doc/library/asyncio-eventloop.rst [hunk 1]
```diff
@@ -1217,32 +1217,52 @@ async/await code consider using the high-level
...
+       error stream to the process' standard output stream
 
    * All other keyword arguments are passed to :class:`subprocess.Popen`
-     without interpretation, except for *bufsize*, *universal_newlines*
-     and *shell*, which should not be specified at all.
+     without interpretation, except for *bufsize*, *universal_newlines*,
+     *shell*, *text*, *encoding* and *errors*, which should not be specified
+     at all.
+
+     The ``asyncio`` subprocess API does not yet support decoding the streams
```
Comment (asvetlov) [inline Doc/library/asyncio-eventloop.rst]: Let's not make assumptions the future in docs.
I mean `subprocess API does not yet support` should be replaced with just `subprocess API does not support`.
I'm open for discussion about text mode support though, please create a separate ussue for that.

**Review by asvetlov — CHANGES_REQUESTED** [review]
The PR is for documentation update, that's nice.
`base_events.py` in `subprocess_exec` and `subprocess_shell` explicitly forbids non-default `universal_newlines`, `shell` and `bufsize`.
It would be nice if `text`, `encoding` and `errors` arguments are explicitly forbidden too by code

**Comment by sbstp** [comment]
> It would be nice if `text`, `encoding` and `errors` arguments are explicitly forbidden too by code

I don't mind writing the code for that, but it seems like there's a chance it will cause breakage if some programs used those params.

**Review by 1st1 — APPROVED** [review]

**Comment by 1st1** [comment]
Please use `blurb` to generate a NEWS entry.

File: Lib/asyncio/base_events.py [hunk 2]
```diff
@@ -1570,6 +1571,10 @@ def _log_subprocess(self, msg, stdin, stdout, stderr):
             raise ValueError("shell must be True")
         if bufsize != 0:
             raise ValueError("bufsize must be 0")
+        if encoding or errors or text:
```
Comment (asvetlov) [inline Lib/asyncio/base_events.py]: Please split this `if` into three separate checks. It makes error messages more specific.

File: Lib/asyncio/base_events.py [hunk 3]
```diff
@@ -1586,13 +1591,19 @@ def _log_subprocess(self, msg, stdin, stdout, stderr):
...
+                              shell=False, bufsize=0,
+                              encoding=None, errors=None, text=None,
+                              **kwargs):
         if universal_newlines:
             raise ValueError("universal_newlines must be False")
         if shell:
             raise ValueError("shell must be False")
         if bufsize != 0:
             raise ValueError("bufsize must be 0")
+        if encoding or errors or text:
```
Comment (asvetlov) [inline Lib/asyncio/base_events.py]: The same

**Review by asvetlov — CHANGES_REQUESTED** [review]
Almost perfect! Please fix last nits

**Comment by sbstp** [comment]
I have made the requested changes; please review again. :)

File: Lib/asyncio/base_events.py [hunk 4]
```diff
@@ -1570,6 +1571,13 @@ def _log_subprocess(self, msg, stdin, stdout, stderr):
             raise ValueError("shell must be True")
         if bufsize != 0:
             raise ValueError("bufsize must be 0")
+        if text:
+            raise ValueError("text must be False")
+        if encoding:
```
Comment (asvetlov) [inline Lib/asyncio/base_events.py]:
```suggestion
        if encoding is not None:
```

File: Lib/asyncio/base_events.py [hunk 5]
```diff
@@ -1570,6 +1571,13 @@ def _log_subprocess(self, msg, stdin, stdout, stderr):
...
+            raise ValueError("encoding must be None")
+        if errors:
```
Comment (asvetlov) [inline Lib/asyncio/base_events.py]:
```suggestion
        if errors is not None:
```

File: Lib/asyncio/base_events.py [hunk 6]
```diff
@@ -1586,13 +1594,22 @@ def _log_subprocess(self, msg, stdin, stdout, stderr):
...
+                              **kwargs):
         if universal_newlines:
             raise ValueError("universal_newlines must be False")
         if shell:
             raise ValueError("shell must be False")
         if bufsize != 0:
             raise ValueError("bufsize must be 0")
+        if text:
+            raise ValueError("text must be False")
+        if encoding:
```
Comment (asvetlov) [inline Lib/asyncio/base_events.py]:
```suggestion
        if encoding is not None:
```

File: Lib/asyncio/base_events.py [hunk 7]
```diff
@@ -1586,13 +1594,22 @@ def _log_subprocess(self, msg, stdin, stdout, stderr):
...
+            raise ValueError("encoding must be None")
+        if errors:
```
Comment (asvetlov) [inline Lib/asyncio/base_events.py]:
```suggestion
        if errors is not None:
```

**Review by asvetlov — CHANGES_REQUESTED** [review]
Please use suggested strict checks to distinguish, e.g. empty string from `None`.
I know that empty string for `errors` is invalid currently but still...

**Review by asvetlov — APPROVED** [review]
Thanks!

**Comment by sbstp** [comment]
Thanks for being patient with me!

**Pull request merged**